mcs/class/corlib/Mono/SafeStringMarshal.cs

```diff
@@ -0,0 +1,46 @@
+//
+// Safe wrapper for a string and its UTF8 encoding
+//
+// Authors:
+//   Aleksey Kliger <aleksey@xamarin.com>
+//   Rodrigo Kumpera <kumpera@xamarin.com>
+//
+// Copyright 2016 Dot net foundation.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace Mono  {
+	internal sealed class SafeStringMarshal : IDisposable {
```

Comment:
Looking at the usage it's infrastructure helper which does not need to pay GC price. If you are worried that someone won't be careful and copy the handle then fair enough